version: "3.9"

services:
  foundry:
    image: "felddy/foundryvtt:release"
    volumes:
      - "./data/foundry:/data"
    environment:
      - FOUNDRY_PASSWORD=${FOUNDRY_PASSWORD}
      - FOUNDRY_USERNAME=${FOUNDRY_USERNAME}
      - FOUNDRY_ADMIN_KEY=${FOUNDRY_KEY}
    restart: always
    healthcheck:
      test: curl --fail http://localhost:30000/ || exit 1
      interval: 5s
      timeout: 3s
      retries: 5
      start_period: 10s

  filebrowser:
    image: "hurlenko/filebrowser:latest"
    volumes:
      - ./data/foundry/:/data
      - ./etc/filebrowser:/config
    restart: always

  nginx:
    image: "jonasal/nginx-certbot:latest"
    environment:
      - NGINX_DOMAIN=${NGINX_DOMAIN}
      - CERTBOT_EMAIL=${NGINX_DOMAIN_EMAIL}
    ports:
      - 80:80
      - 443:443 
    volumes:
      - ./etc/nginx:/etc/nginx/templates
      - ./data/nginx/letsencrypt:/etc/letsencrypt
    depends_on:
      - filebrowser
      - foundry
    restart: always

  backup:
    build: 
      context: ./
      dockerfile: ./etc/backup/Dockerfile
    environment:
      - KOFR_USERNAME=${KOFR_USERNAME}
      - KOFR_PASSWORD=${KOFR_PASSWORD}
    volumes:
      - ./data:/foundry-aio-server/data
      - ./etc:/foundry-aio-server/etc
    depends_on:
      - filebrowser
      - foundry
      - nginx
    restart: always
  
